libraries/protocol/steem_operations.cpp

```diff
@@ -107,37 +107,36 @@ namespace golos { namespace protocol {
         };
 
         void comment_payout_beneficiaries::validate() const {
-            uint32_t sum = 0;
-
-            FC_ASSERT(beneficiaries.size(), "Must specify at least one beneficiary");
-            FC_ASSERT(beneficiaries.size() < 128,
-                      "Cannot specify more than 127 beneficiaries."); // Require size serializtion fits in one byte.
-
-            validate_account_name(beneficiaries[0].account);
-            FC_ASSERT(beneficiaries[0].weight <= STEEMIT_100_PERCENT,
-                      "Cannot allocate more than 100% of rewards to one account");
-            sum += beneficiaries[0].weight;
-            FC_ASSERT(sum <= STEEMIT_100_PERCENT,
-                      "Cannot allocate more than 100% of rewards to a comment"); // Have to check incrementally to avoid overflow
-
-            for (size_t i = 1; i < beneficiaries.size(); i++) {
-                validate_account_name( beneficiaries[i].account);
-                FC_ASSERT(beneficiaries[i].weight <= STEEMIT_100_PERCENT,
-                          "Cannot allocate more than 100% of rewards to one account");
-                sum += beneficiaries[i].weight;
-                FC_ASSERT(sum <= STEEMIT_100_PERCENT,
-                          "Cannot allocate more than 100% of rewards to a comment"); // Have to check incrementally to avoid overflow
-                FC_ASSERT(beneficiaries[i - 1] < beneficiaries[i],
-                          "Benficiaries must be specified in sorted order (account ascending)");
-            }
+            fc::safe<uint32_t> sum = 0;  // avoid overflow
+
+            GOLOS_CHECK_PARAM(beneficiaries, {
+                GOLOS_CHECK_VALUE(beneficiaries.size(), "Must specify at least one beneficiary");
+                GOLOS_CHECK_VALUE(beneficiaries.size() < 128,
+                          "Cannot specify more than 127 beneficiaries."); // Require size serializtion fits in one byte.
+
+                for (auto beneficiar: beneficiaries) {
+                    validate_account_name(beneficiaries[0].account);
+                    GOLOS_CHECK_VALUE(beneficiar.weight <= STEEMIT_100_PERCENT,
+                            "Cannot allocate more than 100% of rewards to one account");
+                    sum += beneficiar.weight;
+                }
+
+                GOLOS_CHECK_VALUE(sum <= STEEMIT_100_PERCENT,
+                        "Cannot allocate more than 100% of rewards to a comment");
```

Comment:
Can you please return this check inside loop. That was simple and elegant solution to avoid any overflows, which required no complex types like `fc::safe<>`

Reply:
Ok. Fixed